### DISCUSSION TITLE
Variable offset for ComponentAlongPath

### DISCUSSION THREAD

**Discussion by gmgopikrishnan** [gdsfactory/gdsfactory#2055] [Q&A] [answered]
Hi All,
Is there an option to have variable offset for the "_gdsfactory.cross_section.ComponentAlongPath_" using a function or list? i.e just like how offset of a section of the cross section can be varied  as a path travels by using a function or list.
Thanks

**Comment by joamatab** [comment, marked as answer]
there is no such as an option at the moment, for that type of unique cases you can write your own
